target/linux/mpc85xx/patches-4.14/104-powerpc-mpc85xx-change-P2020RDB-dts-file-for-OpenWRT.patch

```diff
@@ -0,0 +1,128 @@
+Date: Sun, 30 Dec 2018 23:24:41 +0100
+Subject: [PATCH] powerpc: mpc85xx: change P2020RDB dts file for OpenWRT
+
+This patch apply chages for OpenWRT in P2020RDB
+dts file.
+
+Signed-off-by: Pawel Dembicki <paweldembicki@gmail.com>
+---
+ arch/powerpc/boot/dts/fsl/p2020rdb.dts | 43 +++++++++++++++++++++-----
+ 1 file changed, 35 insertions(+), 8 deletions(-)
+
+diff --git a/arch/powerpc/boot/dts/fsl/p2020rdb.dts b/arch/powerpc/boot/dts/fsl/p2020rdb.dts
+index 435a319958cb..41320958ce98 100644
+--- a/arch/powerpc/boot/dts/fsl/p2020rdb.dts
++++ b/arch/powerpc/boot/dts/fsl/p2020rdb.dts
+@@ -2,6 +2,7 @@
+  * P2020 RDB Device Tree Source
+  *
+  * Copyright 2009-2012 Freescale Semiconductor Inc.
++ * Copyright 2018      Pawel Dembicki <paweldembicki@gmail.com>
+  *
+  * This program is free software; you can redistribute  it and/or modify it
+  * under  the terms of  the GNU General  Public License as published by the
+@@ -9,10 +10,15 @@
+  * option) any later version.
+  */
+ 
++/dts-v1/;
++
++#include <dt-bindings/gpio/gpio.h>
++#include <dt-bindings/input/input.h>
++
+ /include/ "p2020si-pre.dtsi"
+ 
+ / {
+-	model = "fsl,P2020RDB";
++	model = "Freescale P2020RDB";
+ 	compatible = "fsl,P2020RDB";
+ 
+ 	aliases {
+@@ -25,6 +31,17 @@
+ 		pci1 = &pci1;
+ 	};
+ 
++	gpio_export {
++		compatible = "gpio-export";
++		#size-cells = <0>;
++
++		rst_phy_eth2 {
++			gpio-export,name = "rst_phy_eth2";
++			gpio-export,output = <1>;
++			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
++		};
++	};
++
+ 	memory {
+ 		device_type = "memory";
+ 	};
+@@ -56,21 +73,19 @@
+ 			partition@40000 {
+ 				/* 256KB for DTB Image */
+ 				reg = <0x00040000 0x00040000>;
+-				label = "NOR (RO) DTB Image";
+-				read-only;
++				label = "dtb";
+ 			};
+ 
+ 			partition@80000 {
+ 				/* 3.5 MB for Linux Kernel Image */
+ 				reg = <0x00080000 0x00380000>;
+-				label = "NOR (RO) Linux Kernel Image";
+-				read-only;
++				label = "kernel";
+ 			};
+ 
+ 			partition@400000 {
+ 				/* 11MB for JFFS2 based Root file System */
+ 				reg = <0x00400000 0x00b00000>;
+-				label = "NOR (RW) JFFS2 Root File System";
++				label = "rootfs";
+ 			};
+ 
+ 			partition@f00000 {
+@@ -78,9 +93,13 @@
+ 				/* 512KB for u-boot Bootloader Image */
+ 				/* 512KB for u-boot Environment Variables */
+ 				reg = <0x00f00000 0x00100000>;
+-				label = "NOR (RO) U-Boot Image";
++				label = "u-boot";
+ 				read-only;
+ 			};
++			firmware@40000 {
++				reg = <0x00040000 0x00ec0000>;
++				label = "firmware";
++			};
+ 		};
+ 
+ 		nand@1,0 {
+@@ -144,7 +163,15 @@
+ 	soc: soc@ffe00000 {
+ 		ranges = <0x0 0x0 0xffe00000 0x100000>;
+ 
++		gpio0: gpio-controller@fc00 {
++		};
++
+ 		i2c@3000 {
++			adt7461@4c {
++				compatible = "adi,adt7461";
++				reg = <0x4c>;
++			};
++
+ 			rtc@68 {
+ 				compatible = "dallas,ds1339";
+ 				reg = <0x68>;
+@@ -155,7 +182,7 @@
+ 			flash@0 {
+ 				#address-cells = <1>;
+ 				#size-cells = <1>;
+-				compatible = "spansion,s25sl12801", "jedec,spi-nor";
```

Comment:
Interesting. "spansion,s25sl12801" dt compat itself is not registered to any special chip and it seems to exclusively appear in other freescale boards (p1010,p102x, mpc8536, ...) .

However, it does not look like that you are using the spi-nor chip? So, you could just ignore this hunk. The [jedec,spi-nor.txt DT binding text](https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/mtd/jedec%2Cspi-nor.txt) does not necessarily forbid anything other than "jedec,spi-nor".

    May include a device-specific string consisting of the
    manufacturer and name of the chip. A list of supported chip
    names follows.
    Must also include "jedec,spi-nor" for any SPI NOR flash that can
    be identified by the JEDEC READ ID opcode (0x9F)."

Reply:
Ok. I drop my changes.